src/reaper_osara.cpp

```diff
@@ -402,16 +402,38 @@ string formatCursorPosition(TimeFormat format, FormatTimeCacheRequest cache) {
 	return formatTime(GetCursorPosition(), format, false, cache);
 }
 
-const char* formatFolderState(int state, bool reportTrack=true) {
+string formatFolderState(MediaTrack* track) {
+	ostringstream s;
+	int state = (int)GetMediaTrackInfo_Value(track, "I_FOLDERDEPTH");
 	if (state == 0) {
 		// Translators: A track which isn't a folder.
-		return reportTrack ? translate("track") : nullptr;
+		s << translate("track");
+	} else if (state == 1 && GetMediaTrackInfo_Value(track, "P_PARTRACK")) {
+		// Translators: A track which is a nested folder.
+		s << translate("nested folder");
 	} else if (state == 1) {
 		// Translators: A track which is a folder.
-		return translate("folder");
+		s << translate("folder");
+	} else {
+		// Translators: A track which ends its folder.
+		s << translate("end of folder");
+		// find the folder being ended by this track
+		MediaTrack* folderTrack = track;
+		for(int i = 0; i<(-state); ++i) {
```

Comment:
Would this be more readable as something like:

`for (int i = state; i < 0; ++i) {`

Reply:
It probably is more readable. That's actually how I wrote it the first time and then I changed it to count from 0.